android/src/main/kotlin/com/appcues/flutter/FlutterElementTargetingStrategy.kt

```diff
@@ -0,0 +1,158 @@
+
+    override fun evaluateMatch(target: ElementSelector): Int {
+        var weight = 0
+
+        (target as? FlutterElementSelector)?.let {
+            if (!it.identifier.isNullOrEmpty() && it.identifier == identifier) {
+                weight += 1_000
+            }
+        }
+
+        return weight
+    }
+}
+
+internal fun Activity.getParentView(): ViewGroup {
+
+    // try to find the most applicable topmost decorView to capture layout. Typically there is just a single
+    // decorView on the Activity window. However, if something like a dialog modal has been shown, this can add another
+    // window with another decorView on top of the Activity. If we want to support showing content above that layer, we need
+    // to find the topmost decorView like below.
+
+    val decorView = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+        // this is the preferred method on API 29+ with the new WindowInspector function
+        WindowInspector.getGlobalWindowViews().last()
+    } else {
+        @Suppress("SwallowedException", "TooGenericExceptionCaught")
+        try {
+            // this is the less desirable method for API 21-28, using reflection to try to get the root views
+            val windowManagerClass = Class.forName("android.view.WindowManagerGlobal")
+            val windowManager = windowManagerClass.getMethod("getInstance").invoke(null)
+            val getViewRootNames: Method = windowManagerClass.getMethod("getViewRootNames")
+            val getRootView: Method = windowManagerClass.getMethod("getRootView", String::class.java)
+            val rootViewNames = getViewRootNames.invoke(windowManager) as Array<Any?>
+            val rootViews = rootViewNames.map { getRootView(windowManager, it) as View }
+            rootViews.last()
+        } catch (_: Exception) {
+            // if all else fails, use the decorView on the window, which is typically the only one
+            window.decorView
+        }
+    }
+
+    return decorView.rootView as ViewGroup
+}
+
+internal class FlutterElementTargetingStrategy(val plugin: AppcuesFlutterPlugin) : ElementTargetingStrategy {
+
+    private var widgets: MutableMap<String, ViewElement> = mutableMapOf()
+
+    override fun captureLayout(): ViewElement? {
+        return plugin.activity?.getParentView()?.let {
+            val actualPosition = Rect()
+            it.getGlobalVisibleRect(actualPosition)
+
+            val displayMetrics = it.context.resources.displayMetrics
+            val density = displayMetrics.density
+
+            return ViewElement(
+                x = actualPosition.left.toDp(density),
+                y = actualPosition.top.toDp(density),
+                width = actualPosition.width().toDp(density),
+                height = actualPosition.height().toDp(density),
+                selector = null,
+                displayName = null,
+                type = this.javaClass.name,
+                children = widgets.values.toList(),
```

Comment:
not really capturing a true layout hierarchy - just a container with a list of selectable widgets - but enough to suffice for targeting tooltips